cpp/src/utilities/logger.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/utilities/error.hpp>
+#include <cudf/utilities/logger.hpp>
+
+#include "spdlog/sinks/stdout_sinks.h"
+#include <spdlog/sinks/basic_file_sink.h>
+
+#include <string>
+
+namespace {
+
+/**
+ * @brief TODO
+ */
+[[nodiscard]] spdlog::sink_ptr make_libcudf_sink()
+{
+  if (auto filename = std::getenv("LIBCUDF_DEBUG_LOG_FILE"); filename != nullptr) {
+    return std::make_shared<spdlog::sinks::basic_file_sink_mt>(filename, true);
+  } else {
+    return std::make_shared<spdlog::sinks::stdout_sink_mt>();
+  }
```

Comment:
Probably we can use `getenv_or`.

Reply:
good catch. Leftover from the time when this code was in the header and I didn't want to pull in any non-essential headers in.

Reply:
ah, there's one more reason not to use `getenv_or` here. It logs the env var read, and we can't log before setting up the logger :D